### PR TITLE
Add Service Catalog UI Review Revamp

### PR DESCRIPTION
**Pull request by tharikaGitHub**
This PR add the following UI functionality to the ServiceCatalog feature.

1. Adds an Overview page for a Service Entry.
2. Revamp the Onboarding page to suit the Publisher Portal Existing theme.
3. Add option to Create API from a Service Entry.
4. Add functionality to show usages of a particular service entry.
5. Add functionality to download and view a service definition from the UI.
6. Add review revamp fixes.

**Onboarding page for the Service Catalog**
Screenshot 2021-01-05 at 06 04 04

**Current Service Listing Page**
Screenshot 2021-01-09 at 06 32 26

**Service Overview Page**
Screenshot 2021-01-09 at 06 34 58

**Usage view for a service entry**
Screenshot 2021-01-09 at 06 33 28
Screenshot 2021-01-09 at 06 35 10

**Download and View Expandable Service Definition**
Screenshot 2021-01-09 at 06 35 43
Screenshot 2021-01-09 at 06 35 56
Screenshot 2021-01-09 at 06 36 40

**Create API view**
Screenshot 2021-01-09 at 06 37 28
Screenshot 2021-01-09 at 06 37 10

**View for longer service urls**
Screenshot 2021-01-09 at 06 42 06
Screenshot 2021-01-09 at 06 41 25

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+    const {
+        serviceId,
+        history,
+        isOverview,
+        serviceDisplayName,
+        definitionType,
+    } = props;
+    const classes = useStyles();
+    const intl = useIntl();
+    // const [serviceDefinition, setServiceDefinition] = useState(null);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx:127]: Shall we remove this commented code?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+    const [open, setOpen] = useState(false);
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx:130]: shall we remove this extra line?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+                                        helperText={
+                                            (validity.context
+                                                && validity.context.details.map((detail, index) => {
+                                                    return (
+                                                        <div style={{ marginTop: index !== 0 && '10px' }}>
+                                                            {detail.message}
+                                                        </div>
+                                                    );
+                                                }))
+                                            || `API will be exposed in ${actualContext({ context, version })}`
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/CreateApi.jsx:384]: Shouldn't it be 'API will be exposed **with**' a specific context?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/MockResponses.js [hunk 4]
```diff
@@ -31,9 +31,48 @@ class MockResponses {
...
+    * Return sample mocked model data for addSampleService
+    * @export
+    * @returns {*} Mocked Service Catalog model
+    */
+    static addSampleService() {
+        const response = {
+            id: '87624513-0123-6274-9013-012345678901',
+            name: 'Sample-Endpoint',
+            displayName: 'Sample-Endpoint',
+            description: 'A sample Catalog Entry that exposes a REST endpoint',
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/MockResponses.js:66]: Shall we correct this to 'A sample Catalog **entry** that exposes a REST endpoint'

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/ServiceCatalog.js [hunk 5]
```diff
@@ -161,16 +175,19 @@ class ServiceCatalog {
...
+     * @param name {string} Name of the service.
+     * @param version {string} Version of the service.
      * @returns {promise} Service Entry promise.
      */
-    static exportService(id) {
+    static exportService(name, version) {
         const serviceCatalog = new APIClientFactory().getAPIClient(Utils.getCurrentEnvironment(), Utils.CONST.SERVICE_CATALOG_CLIENT)
             .client;
         const promisedServices = serviceCatalog.then(client => {
             // return client.apis['Services'].exportService(
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/ServiceCatalog.js:187]: Shall we remove this commented code segment?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/ServiceCatalog.js [hunk 6]
```diff
@@ -239,6 +256,26 @@ class ServiceCatalog {
...
+    /**
+     * Get the usages of a service
+     * @param id {string} UUID of the service.
+     * @returns {promise} Service Entry API promise.
+     */
+    static getAPIUsages(id) {
+        const serviceCatalog = new APIClientFactory().getAPIClient(Utils.getCurrentEnvironment(), Utils.CONST.SERVICE_CATALOG_CLIENT)
+            .client;
+        const promisedServices = serviceCatalog.then(client => {
+            // return client.apis['Services'].getAPIUsages(
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/ServiceCatalog.js:268]: Shall we remove this commented code segment?

**Comment by erangatl** [comment]
image
Shall we add the following text?  
`All the fields are required`

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/ServiceCatalogComponent.jsx [hunk 7]
```diff
@@ -18,22 +18,29 @@
...
                 exact
                 path='/service-catalog'
                 key={Date.now()}
-                component={Listing}
+                component={(props) => <Listing {...props} />}
+            />
+            <Route
+                path='/service-catalog/:service_uuid/overview'
+                key='/service-catalog/:service_uuid/overview'
+                // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/ServiceCatalogComponent.jsx:39]: Please don't disable the prop spread rule, Pass only the required props,

And in here, we don't need to use a Lambda component here, Just put thee component there and that's it

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/ServiceCatalogComponent.jsx [hunk 8]
```diff
@@ -18,22 +18,29 @@
...
+                component={(props) => <Overview {...props} />}
             />
         </Switch>
     );
 };
 
-export default ServiceCatalog;
+export default ServiceCatalogComponent;
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/ServiceCatalogComponent.jsx:46]: `ServiceCatalogComponent ` is a react component so why we need to append `Component` to the name 😃

**Pull request merged**